tests/e2e/tidbcluster/tidbcluster.go

```diff
@@ -1638,8 +1638,162 @@ var _ = ginkgo.Describe("TiDBCluster", func() {
 			}
 		})
 	})
+
+	ginkgo.Describe("[Feature]: TopologySpreadConstraint", func() {
+		nodeZoneMap := map[string]string{}
+
+		ginkgo.BeforeEach(func() {
+			nodeList, err := c.CoreV1().Nodes().List(metav1.ListOptions{})
+			framework.ExpectNoError(err, "failed to list nodes: %+v")
+			for i := range nodeList.Items {
+				node := &nodeList.Items[i]
+				zone, ok := node.Labels[tests.LabelKeyTestingZone]
+				framework.ExpectEqual(ok, true, "label %s should exist", tests.LabelKeyTestingZone)
+				nodeZoneMap[node.Name] = zone
+			}
+		})
+
+		ginkgo.It("TidbCluster global topology spread contraint for pd", func() {
```

Comment:
What about just deploying one full TC with all the components, we can just wait for the TC ready and check the distribution, what's the benefit to divide them into several cases?

Reply:
I'm afraid the cluster is too large..

Reply:
OK.

Reply:
changed, I find 2 replica is enough to test.